datadog_checks_dev/datadog_checks/dev/tooling/manifest_validator/common/validator.py

```diff
@@ -177,36 +178,81 @@ def validate(self, check_name, decoded, _):
 
 
 class ImmutableAttributesValidator(BaseManifestValidator):
-    """Ensure attributes haven't changed
-    Skip if the manifest is a new file (i.e. new integration)
     """
+    Ensure that immutable attributes haven't changed
+    Skip if the manifest is a new file (i.e. new integration) or if the manifest is being upgraded to V2
+    """
+
+    MANIFEST_VERSION_PATH = "manifest_version"
 
-    FIELDS_NOT_ALLOWED_TO_CHANGE = {
+    IMMUTABLE_FIELD_PATHS = {
         V1: ("integration_id", "display_name", "guid"),
-        V2: ("id", "source_type_name", "app_id"),
+        V2: (
+            "app_id",
+            "app_uuid",
+            "assets/integration/id",
+            "assets/integration/source_type_name",
+        ),
+    }
+
+    SHORT_NAME_PATHS = {
+        V1: (
+            "assets/dashboards",
+            "assets/monitors",
+            "assets/saved_views",
+        ),
+        V2: (
+            "assets/dashboards",
+            "assets/monitors",
+            "assets/saved_views",
+        ),
     }
 
     def validate(self, check_name, decoded, fix):
-        manifest_fields_changed = content_changed(file_glob=f"{check_name}/manifest.json")
+        # Check if previous version of manifest exists
+        # If not, this is a new file so this validation is skipped
+        try:
+            previous = git_show_file(path=f"{check_name}/manifest.json", ref="master")
```

Comment:
For branches that are not up-to-date, wdyta instead using e.g. `https://raw.githubusercontent.com/DataDog/integrations-core/master/<CHECK_NAME>/manifest.json`?

Reply:
I think this is more challenging when we have other repositories that aren't necessarily public 🙂

Reply:
We chatted offline. As suggested, we moved to use `origin/master` so that it uses the remote to stay up to date